002_simulation_structure/002_simulation_structure.tex

```diff
@@ -1,14 +1,108 @@
 \chapter{Simulation Structure}\label{simulation_structure}
 
-\section{Simulation Environment}
+%%%%%%%%
+%%%%%%%% Simulation Environment
+%%%%%%%%
+\section{Simulation Environment}\label{simulation_environment}
 Goes over the backend design
 
-\section{Simulation Flow}
+%%%%%%%%
+%%%%%%%% Simulation Flow
+%%%%%%%%
+\section{Simulation Flow}\label{simulation_flow}
 The order in which events happen
 Need a nice diagram for this
 
-\section{Message Passing}
+%%%%%%%%
+%%%%%%%% Message Passing
+%%%%%%%%
+\section{Message Passing}\label{message_passing}
 Need a nice diagram for this
 
-\section{Health Decay System}
-Need a nice diagram for this
+%%%%%%%%
+%%%%%%%% Health Modeling
+%%%%%%%%
+\section{Health Modeling}\label{health_modeling}
+
+\ToDo{Add diagram with a possible scenario of several days in a row?}
+
+%
+%%%%%%%% Global Description
+%
+\subsection{Global Description}
+
+The health of the agents living in the tower is represented by Health Points (HP). Two mechanisms affect an agent's HP: how much food they eat, and their ``cost of living''. The cost of living represents how many calories a human needs to eat each day to stay healthy. These two mechanisms are implemented using the functions \lstinline$updateHP$ and \lstinline$hpDecay$, respectively. These two functions are described below.
+
+At the end of each day, agents are assigned an HP value based on how much food they have eaten and their cost of living. This HP value is an integer and has a maximum value of \lstinline$MaxHP$, and a minimum value of \lstinline$HPCritical$. As its name suggests, \lstinline$HPCritical$ is a critical HP value for the agents: they can only survive a certain number of days (\lstinline$MaxDayCritical$) at this level. When in the critical state, if agents can increase their HP by \lstinline$HPReqCToW$ (``HP Required to move from Critical To Weak''), then they move into the ``weak state'' (\Cref{fig:health_system}), and their HP takes the value of \lstinline$WeakLevel$. The amount that an agent's HP increases from eating is determined by the function \lstinline$updateHP()$.
+
+\begin{figure}[htb]
+    \centering
+    \includegraphics[width=0.3\linewidth]{002_simulation_structure/images/health_global.pdf}
+    \caption{The health of the agents is represented by a HP value between \lstinline$HPCritical$ and \lstinline$MaxHP$. All HP values which are below \lstinline$WeakLevel$ are classed as critical. The diagram is not drawn to scale.}
+    \label{fig:health_system}
+\end{figure}
+
+\subsection{Food and Health: \lstinline$updateHP$}\label{updateHP}
+To increase their HP, agents need to eat. However, the amount an agent's HP improves can saturate in a single day; eating more than a certain amount will provide an agent with no extra benefit to their HP. Moreover, eating more food will lead to diminishing returns in terms of HP change. Mathematically, the ideas of diminishing returns and saturation are well captured by the step response of a 1st-order system:
```

Comment:
should refer to the equation in the text so change the last line to: "Mathematically, the ideas of diminishing returns and saturation are well captured by the step response of a 1st-order system shown by Equation \ref{updateHP_general}."